compiler/rustc_codegen_llvm/src/builder.rs

```diff
@@ -31,12 +31,135 @@ use tracing::{debug, instrument};
 use crate::abi::FnAbiLlvmExt;
 use crate::attributes;
 use crate::common::Funclet;
-use crate::context::CodegenCx;
+use crate::context::{CodegenCx, SimpleCx};
 use crate::llvm::{self, AtomicOrdering, AtomicRmwBinOp, BasicBlock, False, True};
 use crate::type_::Type;
 use crate::type_of::LayoutLlvmExt;
 use crate::value::Value;
 
+// All Builders must have an llfn associated with them
+#[must_use]
+pub(crate) struct SBuilder<'a, 'll> {
```

Comment:
So my hope was that we could change the regular builder to be

```rust
pub(crate) struct Builder<'a, 'll, CX: Deref<Target = SimpleCx<'ll>>> {
    pub llbuilder: &'ll mut llvm::Builder<'ll>,
    pub cx: CX,
}
```

(may need to use `Borrow` instead of `Deref` and impl it for the context types, because we wanna store `&'a CX`, but then CX won't impl `Deref`)

Once that works, you can have impls for either `impl<CX> Builder<CX>` or `impl Builder<CodegenCx>` depending on what is needed, and the methods would be available for both.

This transition can be simplified a lot by renaming `Builder` to `GenericBuilder` and having a type alias named `Builder`.